packages/cli/src/config/extensions/github.ts

```diff
@@ -61,7 +61,19 @@ export async function cloneFromGit(
       );
     }
 
-    const refToFetch = installMetadata.ref || 'HEAD';
+    let refToFetch = installMetadata.ref;
```

Comment:
Note that the way this works today, if we call `cloneFromGit` when not finding any release assets, we will already be pass in an explicit ref which is the tag name from the latest release.

Are you sure this code is reachable? I am not really quite understanding the exact issue here.